control-plane/p4RuntimeSerializer.cpp

```diff
@@ -1616,10 +1616,15 @@ class P4RuntimeEntriesConverter {
             auto start = simpleKeyExpressionValue(kr->left, typeMap);
             auto end = simpleKeyExpressionValue(kr->right, typeMap);
             if (start == boost::none || end == boost::none) return;
+            // Error on invalid range values
             big_int maxValue = (big_int(1) << keyWidth) - 1;
-            // These should be guaranteed by the frontend
-            BUG_CHECK(*start <= *end, "Invalid range with start greater than end");
-            BUG_CHECK(*end <= maxValue, "End of range is too large");
+            // NOTE: If end value is > max allowed for keyWidth, value gets
+            // wrapped around. A warning is issued in this case by the frontend
+            // earlier.
+            // For e.g. 16 bit key has a max value of 65535, Range of (1..65536)
+            // will be converted to (1..0) and will fail below check.
+            if (*start > *end)
```

Comment:
Maybe you can use an ErrorType argument for ::error.

Reply:
Did try using Errortype initially, but the output for invalid range type adds  a `'..'`, 
`[--Werror=invalid] error: ..: Invalid`
vs current
` error:  Invalid range`

Reply:
It is caused by not using an IR Node in the error message.   See what IR Node you can find in the method and included it in the error message.  Then the IR Node is printed as %1% and your vars are printed with %2% onwards.